test/integration/api/datasets.js

```diff
@@ -813,6 +813,26 @@ describe('datasets and entities', () => {
                   }]);
                 }))));
       }));
+
+      it('should return dataset name only if there is no properties', testService(async (service) => {
+        const asAlice = await service.login('alice', identity);
+
+        await asAlice.post('/v1/projects/1/forms?publish=true')
+          .send(testData.forms.simpleEntity.replace(/entities:saveto[^/]+/g, ''))
+          .set('Content-Type', 'application/xml')
+          .expect(200);
+
+        await asAlice.get('/v1/projects/1/forms/simpleEntity/dataset-diff')
+          .expect(200)
+          .then(({ body }) => {
+            body.should.be.eql([{
+              name: 'people',
+              properties: []
+            }]);
+          });
+
+      }));
+
```

Comment:
Here's another test (that currently fails) for the other part of the issue (a published dataset with no published properties gives a 404 when downloading the CSV):

```js
      it('should let the user download even if if there are no properties', testService(async (service) => {
        const asAlice = await service.login('alice', identity);

        await asAlice.post('/v1/projects/1/forms?publish=true')
          .send(testData.forms.simpleEntity.replace(/entities:saveto[^/]+/g, ''))
          .set('Content-Type', 'application/xml')
          .expect(200);

        await asAlice.get('/v1/projects/1/datasets/people/entities.csv')
          .expect(200)
          .then(({ text }) => {
            text.should.equal('name,label\n');
          });
      }));
```

It's calling `Datasets._getByNameSql`. I tried removing the line `AND ds_properties."publishedAt" IS NOT NULL` from that function but then it lets unpublished properties through. 

I think changing that line to something like this works (but neater):
```sql
AND ((ds_properties."publishedAt" IS NOT NULL AND ds_properties."id" IS NOT NULL) OR ds_properties."id" IS NULL)
```
where if the property exists, it must be published, otherwise it's okay to be null from the left join.